x/wasm/keeper/metrics.go

```diff
@@ -68,6 +73,16 @@ func (p *WasmVMMetricsCollector) Collect(c chan<- prometheus.Metric) {
 	c <- prometheus.MustNewConstMetric(p.CacheElementsDescr, prometheus.GaugeValue, float64(m.ElementsMemoryCache), labelMemory)
 	c <- prometheus.MustNewConstMetric(p.CacheSizeDescr, prometheus.GaugeValue, float64(m.SizeMemoryCache), labelMemory)
 	c <- prometheus.MustNewConstMetric(p.CacheSizeDescr, prometheus.GaugeValue, float64(m.SizePinnedMemoryCache), labelPinned)
+
+	pm, err := p.source.GetPinnedMetrics()
+	if err != nil {
+		return
+	}
```

Comment:
If errors lead to not updating the values and otherwise ignore them (which is fine), can we write the code in a way like:

1. If GetMetrics() succeeds, perform updates. Otherwise skip updates.
1. If GetPinnedMetrics() succeeds, perform updates. Otherwise skip updates.

This extends the old behaviour to more blocks

Reply:
you mean like this? (see latest commit)
Or do you mean: if one of them errors, update nothing

Reply:
Exactly, I like. If no updates are coming to prometheus you can do the debugging. No need for an hard exit here.